app/frontend/app/controllers/group.js

```diff
@@ -0,0 +1,99 @@
+import Ember from 'ember';
+import HasCurrentUser from '../mixins/has-current-user';
+/* global Messenger */
+
+export default Ember.Controller.extend(HasCurrentUser, {
+  // this flag is used so that the "Leave Group" button doesn't switch
+  // to "Join Group" when waiting for the server to respond.
+  loading: false,
+
+  coverUpload: Ember.Object.create(),
+  coverUrl: Ember.computed.any('coverUpload.croppedImage', 'model.coverImage'),
+  coverImageStyle: function() {
+    return "background-image: url(" + this.get('coverUrl') + ")";
+  }.property('coverUrl'),
+
+  currentMember: function() {
+    return this.get('model.members').findBy('user.id', this.get('currentUser.id'));
+  }.property('model.members.@each'),
+
+  isAdmin: function() {
+    return this.get('currentMember') && this.get('currentMember.isAdmin');
+  }.property('currentMember'),
+
+  showEditMenu: false,
+
+  actions: {
+    joinGroup: function(group) {
+      var member = this.store.createRecord('group-member', {
+        groupId: group.get('id'),
+        user: this.get('currentUser.model.content'),
+        pending: true
+      });
+      Messenger().expectPromise(function() {
```

Comment:
Unfortunately not, Messenger calls `.apply` on the promise passed, which overwrites the `this` pointer and the `save()` call fails.